File: Oauth2Simples/src/main/java/br/com/estudos/oauth2/controller/BalanceController.java
package br.com.estudos.oauth2.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.bind.annotation.RequestMapping;

import org.springframework.http.ResponseEntity;
import org.springframework.beans.factory.annotation.Autowired;

import br.com.estudos.oauth2.service.BalanceService;
import br.com.estudos.oauth2.dto.ResumeBalance;

import java.util.List;

@RestController
@CrossOrigin
@RequestMapping(value = "/balance")
public class BalanceController{

     @Autowired
     private BalanceService service;

     @GetMapping(value = "/user-resume")
     public List<ResumeBalance> findSumAllBalanceUser(){
         return service.sumBalanceByTypeBalance();
     }


}

File: kotlin/kotlinEstudos/src/main/java/br/com/kotlin/estudos/controller/BalanceController.kt
package br.com.kotlin.estudos.controller

import br.com.kotlin.estudos.model.dto.balance.BalanceDTORequest
import br.com.kotlin.estudos.model.dto.balance.BalanceResumeResponse
import br.com.kotlin.estudos.model.repository.financial.Balance
import br.com.kotlin.estudos.services.BalanceService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/balance")
class BalanceController(
    private val  balanceService: BalanceService
){
    @GetMapping("/my")
    @ResponseStatus(HttpStatus.OK)
    fun loadStockMarket(): List<BalanceResumeResponse>?{
        println("Entrou balance")
        return balanceService.findByResumeBalance()
    }

    @GetMapping("/historic")
    @ResponseStatus(HttpStatus.OK)
    fun findHistoricByBalance(): List<BalanceResumeResponse>?{
        println("Entrou balance")
        return balanceService.findByResumeBalance()
    }

    @GetMapping
    @ResponseStatus(HttpStatus.OK)
    fun findBalanceByUser(): List<Balance>?{
       return balanceService.findByUsuario()
    }

    @PostMapping
    fun insertBalance(@RequestBody request: BalanceDTORequest){
       balanceService.insertBalance(request)
    }


}